csrc/lower_thread_predicate.h

```diff
@@ -118,7 +139,7 @@ class TORCH_CUDA_CU_API ThreadPredicateMap {
  private:
   // Update the thread_predicates bitset based on provided Expr
   void updateBitSet(const Expr*);
-
+  void avoidConcretizedBroadcastRedundantWrite(const TensorView* out_tv);
```

Comment:
Please write a comment

Reply:
added.
```
// This function is to avoid redundant writes to global memory
// when the tensor has a leaf domain merged from concretized
// broadcast domains and parallelized by thread/block id.
// Only do the write when the index of the leaf domain equals
// write_index_map_.at(pt) where pt is the parallel type.
```